lte/gateway/c/oai/tasks/sctp/sctp_primitives_server.c

```diff
@@ -74,24 +72,51 @@ static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
 
       MessageDef* msg;
 
-      msg = itti_alloc_new_message(TASK_S1AP, SCTP_MME_SERVER_INITIALIZED);
-      SCTP_MME_SERVER_INITIALIZED(msg).successful = true;
-
-      send_msg_to_task(&sctp_task_zmq_ctx, TASK_MME_APP, msg);
+      if (received_message_p->ittiMsg.sctpInit.ppid == S1AP) {
+        msg = itti_alloc_new_message(TASK_S1AP, SCTP_MME_SERVER_INITIALIZED);
+        SCTP_MME_SERVER_INITIALIZED(msg).successful = true;
+        send_msg_to_task(&sctp_task_zmq_ctx, TASK_MME_APP, msg);
+
+      } else if (received_message_p->ittiMsg.sctpInit.ppid == NGAP) {
+        msg = itti_alloc_new_message(TASK_NGAP, SCTP_AMF_SERVER_INITIALIZED);
+        SCTP_AMF_SERVER_INITIALIZED(msg).successful = true;
+        send_msg_to_task(&sctp_task_zmq_ctx, TASK_AMF_APP, msg);
+      } else {
+        OAILOG_ERROR(
+            LOG_SCTP,
+            "received_message_p->ittiMsg.sctpInit.ppid =%d not found ",
+            received_message_p->ittiMsg.sctpInit.ppid);
+      }
     } break;
 
     case SCTP_CLOSE_ASSOCIATION: {
     } break;
 
     case SCTP_DATA_REQ: {
+      uint32_t ppid     = SCTP_DATA_REQ(received_message_p).ppid;
       uint32_t assoc_id = SCTP_DATA_REQ(received_message_p).assoc_id;
       uint16_t stream   = SCTP_DATA_REQ(received_message_p).stream;
       bstring payload   = SCTP_DATA_REQ(received_message_p).payload;
 
-      if (sctpd_send_dl(assoc_id, stream, payload) < 0) {
-        sctp_itti_send_lower_layer_conf(
-            received_message_p->ittiMsgHeader.originTaskId, assoc_id, stream,
-            SCTP_DATA_REQ(received_message_p).mme_ue_s1ap_id, false);
+      if (sctpd_send_dl(ppid, assoc_id, stream, payload) < 0) {
+        if (ppid == S1AP) {
+          OAILOG_DEBUG(
+              LOG_SCTP, "ppid match S1AP in sctp_itti_send_lower_layer_conf ");
```

Comment:
This log is not inside the `sctp_itti_send_lower_layer_conf` function
```suggestion
              LOG_SCTP, "Ppid S1AP in sctp_data_req ");
```